modules/invoicer_invoice/src/Form/InvoicerInvoiceConfigForm.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\invoicer_invoice\Form\InvoicerInvoiceConfigForm.
+ */
+
+namespace Drupal\invoicer_invoice\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Configure file system settings for this site.
+ */
+class InvoicerInvoiceConfigForm extends ConfigFormBase {
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFormId() {
+        return 'invoicer_invoice_config_form';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getEditableConfigNames() {
+        return ['invoicer_invoice.settings'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(array $form, FormStateInterface $form_state) {
+        $config = $this->config('invoicer_invoice.settings');
+
+        $form = array();
+
+        $form['vat_types'] = array(
+            '#type' => 'textarea',
+            '#title' => t('VAT types'),
+            '#size' => 60,
+            '#maxlength' => 60,
+            '#default_value' => implode("\n", $config->get('vat_types')),
```

Comment:
Look at this method on https://api.drupal.org/api/drupal/core!modules!options!src!Plugin!Field!FieldType!ListItemBase.php/class/ListItemBase/8.2.x
```
/**
   * {@inheritdoc}
   */
  public function storageSettingsForm(array &$form, FormStateInterface $form_state, $has_data) {
    $allowed_values = $this->getSetting('allowed_values');
    $allowed_values_function = $this->getSetting('allowed_values_function');

    $element['allowed_values'] = array(
      '#type' => 'textarea',
      '#title' => t('Allowed values list'),
      '#default_value' => $this->allowedValuesString($allowed_values),
      '#rows' => 10,
      '#access' => empty($allowed_values_function),
      '#element_validate' => array(array(get_class($this), 'validateAllowedValues')),
      '#field_has_data' => $has_data,
      '#field_name' => $this->getFieldDefinition()->getName(),
      '#entity_type' => $this->getEntity()->getEntityTypeId(),
      '#allowed_values' => $allowed_values,
    );
............

```

```
And this other:
  /**
   * Generates a string representation of an array of 'allowed values'.
   *
   * This string format is suitable for edition in a textarea.
   *
   * @param array $values
   *   An array of values, where array keys are values and array values are
   *   labels.
   *
   * @return string
   *   The string representation of the $values array:
   *    - Values are separated by a carriage return.
   *    - Each value is in the format "value|label" or "value".
   */
  protected function allowedValuesString($values) {
    $lines = array();
    foreach ($values as $key => $value) {
      $lines[] = "$key|$value";
    }
    return implode("\n", $lines);
  }
```